tests/test_flow_studio_async_performance.py

```
@@ -0,0 +1,72 @@
+            await asyncio.sleep(0.05)
+            current_time = time.time()
+            diff = current_time - last_time
+            if diff > 0.15:
+                raise RuntimeError(f"Event loop blocked for {diff:.3f}s")
+            last_time = current_time
+    except asyncio.CancelledError:
+        # Check one last time on cancellation
+        current_time = time.time()
+        diff = current_time - last_time
+        if diff > 0.15:
+             raise RuntimeError(f"Event loop blocked for {diff:.3f}s (on cancel)")
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)